packages/models-library/src/models_library/osparc_variable_identifier.py

```diff
@@ -83,8 +97,11 @@ def replace_osparc_variable_identifier(  # noqa: C901
         if obj.default_value is not None:
             return deepcopy(obj.default_value)  # type: ignore
     elif isinstance(obj, dict):
-        for key, value in obj.items():
-            obj[key] = replace_osparc_variable_identifier(value, osparc_variables)
+        for key, value in list(obj.items()):
+            del obj[key]
+            obj[
+                replace_osparc_variable_identifier(key, osparc_variables)
```

Comment:
I've accidentally discovered this with a test. So it is tested.

I see no issue with replacing it in keys as well. Since it takes effort to remove it , I would keep it as is.

Reply:
I actually see this quite unsafe to replace keys.  Can you give me an example where this is needed?

Why do you think it takes effort to remove it?
I would say it is quite simple
```python
updates = {}
for key, value in obj.items():
    new_value =  replace_osparc_variable_identifier(value, osparc_variables)
    if new_value != value:
        updates[key] = new_value

```

Reply:
I think that the replacement of keys has the same risks as the replacement of the values has. That's why, in my eyes they are equal. I don't see the difference.

As mentioned above, they are not needed.

Reply:
Do you want me to drop key replacement?

Reply:
OK, let's remove this one.